### PR TITLE
disposal unit now uses a tgui

### PR DESCRIPTION
**Pull request by Mervill**
:cl: Mervill
imageadd: Disposal units now use a tgui instead of plain html
/:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
Lovely! Btw, can I still engage the handle while I am in the process of climbing in, thus allowing me to flush myself? It is akin to pushing the handle down with your foot before jumping in completely.

File: code/modules/recycling/disposal-unit.dm [hunk 1]
```diff
@@ -302,73 +293,50 @@
...
 // handle machine interaction
 
-/obj/machinery/disposal/bin/Topic(href, href_list)
-	if(..())
-		return
-	if(usr.loc == src)
-		usr << "<span class='warning'>You cannot reach the controls from inside!</span>"
+/obj/machinery/disposal/bin/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, \
+									datum/tgui/master_ui = null, datum/ui_state/state = default_state)
+	if(user.loc == src)
```
Comment (AnturK) [inline code/modules/recycling/disposal-unit.dm]: You might want to use the specific ui_state instead here.

File: code/modules/recycling/disposal-unit.dm [hunk 2]
```diff
@@ -302,73 +293,50 @@
...
+		ui = new(user, src, ui_key, "disposal_unit", name, 300, 200, master_ui, state)
+		ui.open()
+
+/obj/machinery/disposal/bin/ui_data(mob/user)
+	var/list/data = list()
+	data["flush"] = flush
+	data["mode"] = mode
+	var/per = Clamp(100* air_contents.return_pressure() / (SEND_PRESSURE), 0, 100)
+	data["per"] = round(per, 1)
+	data["si"] = issilicon(user) && !iscyborg(usr)
```
Comment (AnturK) [inline code/modules/recycling/disposal-unit.dm]: usr => user

**Comment by Niknakflak** [comment]
Just another to lag as I try to open it

**Comment by Mervill** [comment]
@GunHog Yes

File: code/modules/recycling/disposal-unit.dm (hunk 2 above)
Comment (Mervill) [inline code/modules/recycling/disposal-unit.dm]: Good catch

File: code/modules/recycling/disposal-unit.dm (hunk 1 above)
Comment (Mervill) [inline code/modules/recycling/disposal-unit.dm]: I'm not sure what you mean?

File: code/modules/recycling/disposal-unit.dm (hunk 2 above)
Comment (Cyberboss) [inline code/modules/recycling/disposal-unit.dm]: isAI(user)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Incoming5643** [comment]
I agree with niknak, these are too important in a time sensitive situation (self disposaling in a fight) to be laggy.

**Comment by Mervill** [comment]
@Incoming5643 I doubt this will introduce any noticeable latency. Compare to the fire alarm (also a t-gui)

File: code/modules/recycling/disposal-unit.dm (hunk 1 above)
Comment (Mervill) [inline code/modules/recycling/disposal-unit.dm]: are you referring to `datum/ui_state/state = default_state`?

**Comment by Mervill** [comment]
reeeeee (et cetera)

Will resolve when I get home

**Comment by optimumtact** [comment]
The people with toasters from 2001 will be left to die

**Pull request merged**